mva/samples/sample.py

```diff
@@ -1275,16 +1278,60 @@ def records(self,
                     weights_el = reduce(np.multiply, [rec['lep_isele'], rec['lep_0_id_eff_sf_tight']])
                     weights_mu = reduce(np.multiply, [rec['lep_ismu'], rec['lep_0_id_eff_sf_loose']])
                     weights *= reduce(np.add, [weights_el, weights_mu])
-                    
+
                 # drop other weight fields
                 #rec = recfunctions.rec_drop_fields(rec, weight_branches)
                 # add the combined weight
-                rec = recfunctions.rec_append_fields(rec,
+                rec = rec_append_fields(rec,
                     names='weight',
-                    data=weights,
-                    dtypes='f8')
+                    arrs=weights,
+                    dtypes=np.dtype('f8'))
                 if rec['weight'].shape[0] > 1 and rec['weight'].sum() == 0:
                     log.warning("{0}: weights sum to zero!".format(table.name))
+            mom_branches = [
```

Comment:
Why does this calculation happen twice (in data.py and sample.py) ? 
Couldn't it be done once in sample.py in a dedicated method ?